WordPress/build.gradle

```diff
@@ -117,6 +117,11 @@ android {
         // MPAndroidChart uses androidX - remove this line when we migrate everything to androidX
         exclude 'META-INF/proguard/androidx-annotations.pro'
     }
+
```

Comment:
OK, so this is obviously the big comment, which is why I'm coming to replying to it last of the lot. I had to think about this a bit.

OK, so you're right that it's a new architectural direction in a sense, but not a prescriptive one. By this I mean the decision to use Android view binding would not force the update of any parts of the application which were not already using it. It simply creates files in the build directory using the preprocessor which are ignored by anything which doesn't use them.
I would therefore really really like to include its usage. As you can see, it drops the number of lines of non-declarative and not easily re-usable code by a large margin. I do completely understand the objection of using mixed arch in the same view, but if this is the _only_ problem then I would actually much rather refactor the existing than let the existing determine what should be used for the new.

_Having said all of the above_; if you're saying that you'd rather I did not attempt to introduce an architectural change at all, then I can write it the old-fashioned way, but being completely honest, I'd rather not. I actually think including view bindings can produce the cleanest arch you can get for Android, and is not a third party library for which you'd have to consider support being dropped etc. If it is something that would be considered realistically, then I'm happy to write the p2 post explaining it and provide a strong case for why I think it's a good idea with code examples.

Reply:
This discussion is starting to have two parts
1. Should we use data binding in our codebase?

I'm not against it in general. I have a couple of thoughts about it, but I've used it in a large project and it worked pretty well. I'd rather not go into details in this thread as it could easily get out of hand. Having said that, I'll be more than happy to continue with this discussion on a p2 post/public slack channel.

2. Should we introduce data binding in this PR?

This is the part I feel strongly about. I don't think introducing data binding should be our decision, but rather a decision of the whole team (or several members at least). 

>you're right that it's a new architectural direction in a sense, but not a prescriptive one.

Yes, nothing forces you to use it everywhere, but at the same time it brings yet another inconsistency into our code base. Moreover, I believe it increases the build time and has other side effect which I feel should be considered and discussed first.

Let's ask a third person about their thoughts since we apparently don't share the same opinion, which is completely fine. @oguzkocer Would you mind chiming in? Thanks!

Reply:
I agree with @malinajirka that we shouldn't just throw in a dependency without giving other developers a chance to comment on it. Whether data-binding is the right way to go or not is not really the point of this discussion, so my opinion on that part doesn't really matter. However, for some context, I know that we tried data-binding a while back and had some issues with it, so we decided to put it off. I imagine those issues would already be addressed, so there is probably a good case to made for it, but I'd prefer this discussion to happen outside of a PR.

@onepointsixtwo If you feel strongly about data binding and more importantly if you don't feel comfortable with proposing this PR without it, please feel free to start a discussion for it while we put this PR on hold. Otherwise, I suggest making the necessary changes to this PR _and then_ starting the discussion for data-binding so we can keep things going.

Outside of the discussion of data-binding, I think it's important that we are all able to work within the constraints and follow patterns even if we don't feel comfortable with them. In a code base like WordPress, I can't stress the importance of that enough.

Hope this helps!